sabnzbd/interface.py

```diff
@@ -150,25 +167,43 @@ def internal_wrap(*args, **kwargs):
     return internal_wrap
 
 
-def check_access(access_type=4):
+def check_access(access_type: int = 4, warn_user: bool = False) -> bool:
     """Check if external address is allowed given access_type:
     1=nzb
     2=api
     3=full_api
     4=webui
     5=webui with login for external
     """
-    referrer = cherrypy.request.remote.ip
+    # Easy, it's allowed
+    if access_type <= cfg.inet_exposure():
+        return True
 
     # CherryPy will report ::ffff:192.168.0.10 on dual-stack situation
-    # It will always contain that ::ffff: prefix
-    range_ok = not cfg.local_ranges() or bool(
-        [1 for r in cfg.local_ranges() if (referrer.startswith(r) or referrer.replace("::ffff:", "").startswith(r))]
-    )
-    allowed = referrer in ("127.0.0.1", "::ffff:127.0.0.1", "::1") or range_ok or access_type <= cfg.inet_exposure()
-    if not allowed:
-        logging.debug("Refused connection from %s", referrer)
-    return allowed
+    # It will always contain that ::ffff: prefix, the ipaddress module can handle that
+    referrer = cherrypy.request.remote.ip
```

Comment:
To avoid confusion, you might want to change the variable name to something other than "referrer" - as that usually means the (user-set) http header.